src/main/java/rx/internal/operators/OnSubscribeTimeoutTimedWithFallback.java

```diff
@@ -0,0 +1,230 @@
+import rx.Scheduler.Worker;
+import rx.functions.Action0;
+import rx.internal.producers.ProducerArbiter;
+import rx.internal.subscriptions.SequentialSubscription;
+import rx.plugins.RxJavaHooks;
+
+/**
+ * Switches to consuming a fallback Observable if the main source doesn't signal an onNext event
+ * within the given time frame after subscription or the previous onNext event.
+ *
+ * @param <T> the value type
+ * @since 1.3.3
+ */
+public final class OnSubscribeTimeoutTimedWithFallback<T> implements Observable.OnSubscribe<T> {
+
+    final Observable<T> source;
+
+    final long timeout;
+
+    final TimeUnit unit;
+
+    final Scheduler scheduler;
+
+    final Observable<? extends T> fallback;
+
+    public OnSubscribeTimeoutTimedWithFallback(Observable<T> source, long timeout,
+            TimeUnit unit, Scheduler scheduler,
+            Observable<? extends T> fallback) {
+        this.source = source;
+        this.timeout = timeout;
+        this.unit = unit;
+        this.scheduler = scheduler;
+        this.fallback = fallback;
+    }
+
+    @Override
+    public void call(Subscriber<? super T> t) {
+        TimeoutMainSubscriber<T> parent = new TimeoutMainSubscriber<T>(t, timeout, unit, scheduler.createWorker(), fallback);
+        t.add(parent.upstream);
+        t.setProducer(parent.arbiter);
+        parent.startTimeout(0L);
+        source.subscribe(parent);
+    }
+
+    static final class TimeoutMainSubscriber<T> extends Subscriber<T> {
+
+        final Subscriber<? super T> actual;
+
+        final long timeout;
+
+        final TimeUnit unit;
+
+        final Worker worker;
+
+        final Observable<? extends T> fallback;
+
+        final ProducerArbiter arbiter;
+
+        final AtomicLong index;
+
+        final SequentialSubscription task;
+
+        final SequentialSubscription upstream;
+
+        long consumed;
+
+        TimeoutMainSubscriber(Subscriber<? super T> actual, long timeout,
+                TimeUnit unit, Worker worker,
+                Observable<? extends T> fallback) {
+            this.actual = actual;
+            this.timeout = timeout;
+            this.unit = unit;
+            this.worker = worker;
+            this.fallback = fallback;
+            this.arbiter = new ProducerArbiter();
+            this.index = new AtomicLong();
+            this.task = new SequentialSubscription();
+            this.upstream = new SequentialSubscription(this);
+            this.add(worker);
+            this.add(task);
+        }
+
+
+        @Override
+        public void onNext(T t) {
+            long idx = index.get();
+            if (idx == Long.MAX_VALUE) {
+                return;
+            }
+            if (!index.compareAndSet(idx, idx + 1)) {
+                return;
+            }
+
+            Subscription s = task.get();
+            if (s != null) {
+                s.unsubscribe();
+            }
+
+            actual.onNext(t);
+
+            consumed++;
+
+            startTimeout(idx + 1);
```

Comment:
If I'm reading this correctly, I think there may be a slight behavior difference between this implementation and the existing one.  The existing implementation will timeout (and thus switch) if there are 0 calls to `onNext` before the timeout.  Here I think there needs to be at least 1 call before the timer is set.  I'll confirm this and send a test case when I get a chance.

Turns out we actually want both semantics.  We're using the other timeout variant mentioned below to get the "have to have at least one message" semantic by having `firstTimeoutSelector` return `Observable.never()`.  If there were a way to get both behaviors with just this operator, that would great.

Reply:
Never mind -- I see the `startTimeout` call in `call` above.

Reply:
With this operator I can override `call` to make that optional and get the 2 semantics I'm interested in, so this will be perfect.  Thanks.

Reply:
Except that it's `final`, doh!.  I'm assuming that's a standard policy for operators.  In that case would it be possible to add a `boolean` parameter to the constructor where `true` means start the timer on subscribe and `false` means don't?  Understood if that seems too specialized, but it would save us having to use the more complex operator.

Reply:
Sorry, no more new features for 1.x, only bugfixes.

Reply:
Understood.